bwc-test/build.gradle

```diff
@@ -69,7 +69,7 @@ repositories {
 dependencies {
     testImplementation 'com.google.guava:guava:30.0-jre'
     testImplementation "org.opensearch.test:framework:${opensearch_version}"
-    testImplementation 'org.apache.logging.log4j:log4j-core:2.17.1'
+    testImplementation 'org.apache.logging.log4j:log4j-core:2.20.0'
```

Comment:
Updated this to match the version in core to try to resolve the issue in bwc test. I'm not sure why this dependency is required here or if the version from core's buildSrc/version.properties can be referenced in this build.gradle file. This is a separate gradle build file from the one at the top level.

Reply:
@peternied @willyborankin Are either of you available to help diagnose the issue with bwc test on this PR? I'm not very familiar with testclusters and not sure if any change needs to be done here or another branch to get those checks working.

Reply:
The root cause is:

```
»  java.lang.NoClassDefFoundError: org/opensearch/core/common/settings/SecureString
»  	at org.opensearch.security.ssl.SecureSSLSettings$InsecureFallbackStringSetting.<init>(SecureSSLSettings.java:113)
»  	at org.opensearch.security.ssl.SecureSSLSettings$SSLSetting.asSetting(SecureSSLSettings.java:78)
»  	at org.opensearch.security.ssl.SecureSSLSettings.lambda$getSecureSettings$0(SecureSSLSettings.java:101)
»  	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
»  	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
»  	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
»  	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
»  	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
»  	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
»  	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
»  	at org.opensearch.security.ssl.SecureSSLSettings.getSecureSettings(SecureSSLSettings.java:102)
```

but I can't tell why that class isn't found because it exists on the main branch of core. It does not exist on 2.x, but in 2.x it uses the correct import for that class.

Reply:
It looks like its not using the HEAD of main for core when building the testcluster based off main

Reply:
Regarding BWC it does not use OS versioning for log4j ... I missed it when prepared fix in other places

Reply:
@cwperks hope it will fix https://github.com/opensearch-project/security/pull/3004

Reply:
@willyborankin I invited you to be a collaborator on my fork if you want to push changes directly to this PR.

Reply:
Does this line in the bwc test workflow mean its checking out main instead of the PR branch? https://github.com/opensearch-project/security/blob/main/.github/workflows/bwc-tests.yml#L41

Reply:
@cwperks now it fails since it take source code and compile it. If i understand gradle script. So theoretically after merging of this PR it should start building the plugin :-)

Reply:
> Does this line in the bwc test workflow mean its checking out main instead of the PR branch? https://github.com/opensearch-project/security/blob/main/.github/workflows/bwc-tests.yml#L41

yes it is. thats why i think if we merge your changes most probably it will fix BWC since previous problems were related to libraries only